File: checkout-core/src/main/java/com/adyen/checkout/core/model/Item.java
/*
 * Copyright (c) 2017 Adyen N.V.
 *
 * This file is open source and available under the MIT license. See the LICENSE file for more info.
 *
 * Created by timon on 05/08/2017.
 */

package com.adyen.checkout.core.model;

import android.os.Parcelable;
import android.support.annotation.NonNull;

import com.adyen.checkout.base.TxSubVariantProvider;

public interface Item extends TxSubVariantProvider, Parcelable {
    /**
     * @return The ID of the {@link Item}.
     */
    @NonNull
    String getId();

    /**
     * @return The display name of the {@link Item}.
     */
    @NonNull
    String getName();
}

File: core/src/main/java/com/adyen/checkout/core/components/data/model/Item.kt
/*
 * Copyright (c) 2019 Adyen N.V.
 *
 * This file is open source and available under the MIT license. See the LICENSE file for more info.
 *
 * Created by caiof on 27/5/2019.
 */
package com.adyen.checkout.core.components.data.model

import com.adyen.checkout.core.common.internal.model.ModelObject
import com.adyen.checkout.core.common.internal.model.getStringOrNull
import kotlinx.parcelize.Parcelize
import org.json.JSONObject

@Parcelize
data class Item(
    val id: String? = null,
    val name: String? = null,
) : ModelObject() {

    companion object {
        private const val ID = "id"
        private const val NAME = "name"

        @JvmField
        val SERIALIZER: Serializer<Item> = object : Serializer<Item> {
            override fun serialize(modelObject: Item): JSONObject {
                return JSONObject().apply {
                    putOpt(ID, modelObject.id)
                    putOpt(NAME, modelObject.name)
                }
            }

            override fun deserialize(jsonObject: JSONObject): Item {
                return Item(
                    id = jsonObject.getStringOrNull(ID),
                    name = jsonObject.getStringOrNull(NAME),
                )
            }
        }
    }
}
